bridge-sdk/eth-proof/src/error.rs

```diff
@@ -0,0 +1,18 @@
+use cita_trie::TrieError;
+use crate::eth_rpc_client::EthRpcError;
+
+#[derive(thiserror::Error, Debug)]
+#[error("Failed to generate Ethereum proof: {0}")]
+pub struct EthProofError(pub String);
```

Comment:
It is not a good approach to unifying the errors, the crate user should be able to distinguish between the errors to retry on `EthRpcError` and skip on `TrieError`.
So you need just to use enum that combine those two types of errors.

Reply:
Wasn't sure which level of detail we want to have here. Converted as you suggested